topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlHandler.java

```diff
@@ -0,0 +1,205 @@
+import static java.lang.String.format;
+import static org.apache.logging.log4j.core.util.Throwables.getRootCause;
+
+/**
+ * Handler for Cruise Control requests.
+ */
+public class CruiseControlHandler {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(CruiseControlHandler.class);
+    
+    private final TopicOperatorConfig config;
+    private final TopicOperatorMetricsHolder metricsHolder;
+    private final CruiseControlClient cruiseControlClient;
+
+    /**
+     * Create a new instance.
+     *
+     * @param config Topic Operator configuration.
+     * @param metricsHolder Metrics holder.
+     * @param cruiseControlClient Cruise Control client.
+     */
+    public CruiseControlHandler(TopicOperatorConfig config,
+                                TopicOperatorMetricsHolder metricsHolder,
+                                CruiseControlClient cruiseControlClient) {
+        this.config = config;
+        this.metricsHolder = metricsHolder;
+        this.cruiseControlClient = cruiseControlClient;
+    }
+
+    /**
+     * Send a topic_configuration request to create a task for replication factor change of one or more topics.
+     * This should be called when one or more .spec.replicas changes are detected.
+     * Note that this method also updates the KafkaTopic status.
+     * 
+     * @param reconcilableTopics Pending replicas changes.
+     * @return Results with status updates.
+     */
+    public Results requestPendingChanges(List<ReconcilableTopic> reconcilableTopics) {
+        var results = new Results();
+        if (reconcilableTopics.isEmpty()) {
+            return results;
+        }
+        results.addAll(updateToPending(reconcilableTopics, "Replicas change pending"));
+
+        var timerSample = TopicOperatorUtil.startExternalRequestTimer(metricsHolder, config.enableAdditionalMetrics());
+        try {
+            LOGGER.debugOp("Sending topic configuration request, topics {}", topicNames(reconcilableTopics));
+            var kafkaTopics = reconcilableTopics.stream().map(ReconcilableTopic::kt).collect(Collectors.toList());
+            var userTaskId = cruiseControlClient.topicConfiguration(kafkaTopics);
+            results.addAll(updateToOngoing(reconcilableTopics, "Replicas change ongoing", userTaskId));
+        } catch (Throwable t) {
+            results.addAll(updateToFailed(reconcilableTopics, format("Replicas change failed, %s", getRootCause(t).getMessage())));
+        }
+        TopicOperatorUtil.stopExternalRequestTimer(timerSample, metricsHolder::cruiseControlTopicConfig, config.enableAdditionalMetrics(), config.namespace());
+        
+        return results;
+    }
+
+    /**
+     * Send a user_tasks request to check the state of ongoing replication factor changes.
+     * This should be called periodically to update the active tasks cache and KafkaTopic status.
+     * Note that this method also updates the KafkaTopic status.
```

Comment:
Similar comment

Reply:
I forgot to remove the note, these pending status updates are now also stored in Results.